python/lsst/analysis/tools/actions/plot/histPlot.py

```diff
@@ -216,56 +270,118 @@ def _assignColors(self):
+            if panel_range[1] - panel_range[0] == 0:
+                log.info(
+                    "NOTE: panel_range for {} based on med/sigMad was 0.  Computing using "
+                    "percentile range instead.".format(panel)
+                )
+                panel_range = self._getPercentilePanelRange(data, panel)
+
         for i, hist in enumerate(self.panels[panel].hists):
             hist_data = data[hist][np.isfinite(data[hist])]
             ax.hist(
                 hist_data,
                 range=panel_range,
                 bins=self.panels[panel].bins,
                 histtype="step",
+                density=self.panels[panel].histDensity,
                 lw=2,
-                color=col[i],
+                color=colors[i],
                 label=self.panels[panel].hists[hist],
             )
-            num, med, mad = self._calcStats(hist_data)
-            nums.append(num)
-            meds.append(med)
-            mads.append(mad)
-            ax.axvline(med, ls="--", lw=1, c=col[i])
-        ax.legend(fontsize=6, loc="upper left")
+            ax.axvline(meds[i], ls=(0, (5, 3)), lw=1, c=colors[i])
+
+        ax.legend(fontsize=legend_font_size, loc="upper left", frameon=False)
         ax.set_xlim(panel_range)
-        ax.set_xlabel(self.panels[panel].label)
+        # The following accommodates spacing for ranges with large numbers
+        # but small-ish dynamic range (example use case: RA 300-301).
+        if ncols > 1 and max(np.abs(panel_range)) >= 100 and (panel_range[1] - panel_range[0]) < 5:
+            ax.xaxis.set_major_formatter("{x:.2f}")
+            ax.tick_params(axis="x", labelrotation=25, pad=-1)
+        ax.set_xlabel(self.panels[panel].label, fontsize=label_font_size)
+        y_label = "Normalized (PDF)" if self.panels[panel].histDensity else "Frequency"
+        ax.set_ylabel(y_label, fontsize=label_font_size)
         ax.set_yscale(self.panels[panel].yscale)
-        ax.tick_params(labelsize=7)
+        ax.tick_params(labelsize=max(5, label_font_size - 2))
         # add a buffer to the top of the plot to allow headspace for labels
         ylims = list(ax.get_ylim())
         if ax.get_yscale() == "log":
             ylims[1] = 10 ** (np.log10(ylims[1]) * 1.1)
         else:
             ylims[1] *= 1.1
         ax.set_ylim(ylims[0], ylims[1])
+
+        # Draw a vertical line at expected value, if given.  If histDensity
+        # is True, also plot a reference PDF with mean = expected value and
+        # sigma = 1 for reference.
+        if self.panels[panel].expectedValue is not None:
+            ax2 = ax.twinx()
+            ax2.axis("off")
+            ax2.set_xlim(ax.get_xlim())
+            ax2.set_ylim(ax.get_ylim())
+
+            if self.panels[panel].histDensity:
+                expected_label = None
+            else:
+                expected_label = "${{\\mu_{{expected}}}}$: {}".format(self.panels[panel].expectedValue)
```

Comment:
Why `mu_expected`? If I want to add a generic reference line at RA=300 deg, I think it might be confusing to have this black line labeled as `mu_expected`. Does this need a string label at all, i.e., will just the line and the value suffice?

Reply:
As head space at the top of these panels is in high demand, would `mu_ref` work for you? Not a deal breaker if not, but I think syncs nicely with `P_norm` opposite.